ui/cypress/e2e/explorer.test.ts

```diff
@@ -34,6 +34,73 @@ describe('DataExplorer', () => {
     })
   })
 
+  describe('numeric input using custom bin sizes in Histograms', () => {
```

Comment:
Check out [Cypress - best practices](https://docs.cypress.io/guides/references/best-practices.html#Creating-%E2%80%9Ctiny%E2%80%9D-tests-with-a-single-assertion) around a number of tests with single assertions (I'm not really a fan of the whole "you were told this and you followed it" tone, but whatever):

## Creating “tiny” tests with a single assertion

While technically this runs fine - this is really excessive, and not performant.

Why you did this pattern in unit tests:

    When assertions failed you relied on the test’s title to know what failed
    You were told that adding multiple assertions was bad and accepted this as truth
    There was no performance penalty splitting up multiple tests because they run really fast

Why you shouldn’t do this in Cypress:

    Writing integration tests is not the same as unit tests
    You will always know (and can visually see) which assertion failed in a large test
    Cypress runs a series of async lifecycle events that reset state between tests
    Resetting tests is much slower than adding more assertions

It is common for tests in Cypress to issue 30+ commands. Because nearly every command has a default assertion (and can therefore fail), even by limiting your assertions you’re not saving yourself anything because any single command could implicitly fail.

---

I think it's probably better to try to have this run in a single test, or as few tests as possible, if it makes sense. But like the docs say - "it's fine..." and if you want to leave it this way, feel free.

Reply:
Good point. I have compacted the tests. Please re-review.